### PR TITLE
Joomla CMS [#27499] Categories deletion or status changed return only 1 category notice 

### PR DESCRIPTION
**Pull request by oc666**
The pull request fix the publish method of all tables and affected by classes
http://joomlacode.org/gf/project/joomla/tracker/?action=TrackerItemEdit&tracker_item_id=27499

### CODE REVIEW

**Comment by oc666** [comment]
Hey,
Where can I see the failures?
Thanks

**Comment by elkuku** [comment]
Hey,
the Joomla! project unfortunately does not provide a detailed error analysis (yet).

You should set up the CodeSniffer (and PHPUnit) on your local system.
Detailed instructions can be found here: http://docs.joomla.org/Joomla_CodeSniffer

As a **personal unofficial** playground project I set up my own instance of the script that Mr. joomla-jenkins uses for his analysis and tweaked a HTML output, e.g. for your pull request:
http://elkuku.github.com/pulltester/pulls/730

hF

File: libraries/joomla/application/component/modeladmin.php [hunk 1]
```diff
@@ -862,7 +862,8 @@ public function publish(&$pks, $value = 1)
 		}
 
 		// Attempt to change the state of the records.
-		if (!$table->publish($pks, $value, $user->get('id')))
+		if (($affectedRows = $table->publish($pks, $value, $user->get('id')))
+                                === FALSE)
```
Comment (elinw) [inline libraries/joomla/application/component/modeladmin.php:866]: Don't have all these extra tabs.

File: libraries/joomla/database/tablenested.php [hunk 2]
```diff
@@ -983,6 +984,8 @@ public function publish($pks = null, $state = 1, $userId = 0)
 				$e = new JException(JText::sprintf('JLIB_DATABASE_ERROR_PUBLISH_FAILED', get_class($this), $this->_db->getErrorMsg()));
 				$this->setError($e);
 				return false;
+			} else {
+				$affectedRows += $this->_db->getAffectedRows();
```
Comment (elinw) [inline libraries/joomla/database/tablenested.php:988]: }
else
{

**Comment by elinw** [comment]
I think what's happening is that it is returning the number of checked items/branches not the number of changed items.

**Comment by chdemko** [comment]
@oc666 Any news on this?

**Comment by elinw** [comment]
Christophe.,

I'm still waiting for anhyone to reply to my question about this on the platform list.

**Comment by LouisLandry** [comment]
This has been sitting for over 2 months without update and is not currently mergeable.  I'm going to close it for now.  If you want to continue the discussion please rebase the branch against current master and re-open the pull request.  Cheers.

**Comment by elinw** [comment]
@Louis,

No one ever responded to the question about expecte behavior that I asked on the platform list which is why we couldn't move forward. It would be great to get a reply on that.

**Comment by piotr-cz** [comment]
@elinw Please add a link to the topic on platform group

**Pull request closed without merging**